src/pair.ts

```diff
@@ -31,12 +31,20 @@ export abstract class Pair {
 		swappaPairAddress: Address,
 	}>;
 	public abstract refresh(): Promise<void>;
-	public swapData(inputToken: Address): SwapData {
+	public swapData(inputToken: Address, inputAmount?: BigNumber): SwapData {
```

Comment:
optionally accept `inputAmount` and encode as a `uint256` at the end of the existing swap data bytes